src/changelog.rs

```diff
@@ -0,0 +1,64 @@
+use crate::*;
+use std::collections::VecDeque;
+
+#[derive(Clone)]
+#[near(serializers=[borsh, json])]
+pub struct ChangeLog {
+    pub block_id: u64,
+    pub block_timestamp: u64,
```

Comment:
There are dedicated type aliases in near-sdk-rs for BlockHeight and BlockTimestamp, so I would use those.